pkg/tbtc/node.go

```diff
@@ -54,11 +56,14 @@ type node struct {
 	// signingExecutors is the cache holding signing executors for specific wallets.
 	// The cache key is the uncompressed public key (with 04 prefix) of the wallet.
 	signingExecutors map[string]*signingExecutor
+
+	walletDispatcher *walletDispatcher
 }
```

Comment:
Let's change the order of fields in `node struct` and add some docs. Figuring out which component is responsible for what is getting complicated now.

```
// node represents the current state of an ECDSA node.
type node struct {
	groupParameters *GroupParameters

	chain          Chain
	btcChain       bitcoin.Chain
	netProvider    net.Provider
	walletRegistry *walletRegistry

	// walletDispatcher ensures only one action is executed by a wallet at
	// a time. All possible activities of a created wallet must be represented
	// by appropriate actions dispatched through this component.
	walletDispatcher *walletDispatcher

	// protocolLatch makes sure no expensive number generator operations are
	// running when signing or generating a wallet key are executed. The
	// protocolLatch is used by dkgExecutor and signingExecutor.
	protocolLatch *generator.ProtocolLatch

	// dkgExecutor encapsulates the logic of distributed key generation.
	//
	// dkgExecutor MUST NOT be used outside of this struct.
	dkgExecutor *dkgExecutor

	signingExecutorsMutex sync.Mutex
	// signingExecutors is the cache holding signing executors for specific wallets.
	// The cache key is the uncompressed public key (with 04 prefix) of the wallet.
	// signingExecutor encapsulates the generic logic of signing messages.
	//
	// signingExecutors MUST NOT be used outside of this struct. Please use
	// wallet actions and walletDispatcher to execute an action on an existing
	// wallet.
	signingExecutors map[string]*signingExecutor
}
```

Reply:
Holding off with this one until we come to an agreement in https://github.com/keep-network/keep-core/pull/3521#discussion_r1173652203